browser/ai_chat/tools/click_tool.cc

```diff
@@ -0,0 +1,133 @@
+
+namespace ai_chat {
+
+ClickTool::ClickTool(ContentAgentTaskProvider* task_provider,
+                     actor::ActorKeyedService* actor_service)
+    : actor_service_(actor_service), task_provider_(task_provider) {}
+
+ClickTool::~ClickTool() = default;
+
+std::string_view ClickTool::Name() const {
+  return "click_element";
+}
+
+std::string_view ClickTool::Description() const {
+  return "Click on an element in the current web page. Use the 'target' "
+         "object to specify either DOM element identifiers or screen "
+         "coordinates. Supports left/right click and single/double click.";
+}
+
+std::optional<base::Value::Dict> ClickTool::InputProperties() const {
+  return CreateInputProperties(
+      {{"target", target_util::TargetProperty("Element to click on")},
+       {"click_type",
+        StringProperty("Type of click to perform",
+                       std::vector<std::string>{"left", "right"})},
+       {"click_count",
+        StringProperty("Number of clicks to perform",
+                       std::vector<std::string>{"single", "double"})}});
+}
+
+std::optional<std::vector<std::string>> ClickTool::RequiredProperties() const {
+  return std::optional<std::vector<std::string>>(
+      {"target", "click_type", "click_count"});
+}
+
+void ClickTool::UseTool(const std::string& input_json,
+                        UseToolCallback callback) {
+  auto input = base::JSONReader::ReadDict(input_json);
+
+  if (!input.has_value()) {
+    std::move(callback).Run(CreateContentBlocksForText(
+        "Failed to parse input JSON. Please try again."));
+    return;
+  }
+
+  // Validate click_type and click_count
+  const auto* click_type = input->FindString("click_type");
+  const auto* click_count = input->FindString("click_count");
+
+  if (!click_type || (*click_type != "left" && *click_type != "right")) {
+    std::move(callback).Run(CreateContentBlocksForText(
+        "Invalid or missing click_type. Must be 'left' or 'right'."));
+    return;
+  }
+
+  if (!click_count || (*click_count != "single" && *click_count != "double")) {
+    std::move(callback).Run(CreateContentBlocksForText(
+        "Invalid or missing click_count. Must be 'single' or 'double'."));
+    return;
+  }
+
+  // Extract and parse target object
+  const base::Value::Dict* target_dict = input->FindDict("target");
+  if (!target_dict) {
+    std::move(callback).Run(
+        CreateContentBlocksForText("Missing 'target' object"));
+    return;
+  }
+
+  auto target = target_util::ParseTargetInput(*target_dict);
+  if (!target.has_value()) {
+    std::move(callback).Run(CreateContentBlocksForText(target.error()));
+    return;
+  }
+
+  task_provider_->GetOrCreateTabHandleForTask(
+      base::BindOnce(&ClickTool::OnTabHandleCreated,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback),
+                     std::move(target.value()), *click_type, *click_count));
```

Comment:
I think maybe we should be `std::move`ing out of `click_type` rather than copying